src/script/script.h

```diff
@@ -224,7 +219,7 @@ class CScriptNum
                         const size_t nMaxNumSize = nDefaultMaxNumSize)
     {
         if (vch.size() > nMaxNumSize) {
-            throw scriptnum_error("script number overflow");
+            throw SCRIPT_ERR_NUMOVERFLOW;
```

Comment:
Throwing a bare enum (or anything not deriving from `std::exception`)  is a bit strange in C++. I think if you want to add an error code it'd be better to add an error code field to the `scriptnum_error` class and its constructor, and keep throwing that.